openwisp_controller/config/admin.py

```diff
@@ -63,6 +68,15 @@ def add_view(self, request, form_url='', extra_context=None):
         extra_context = self.get_extra_context()
         return super(DeviceAdmin, self).add_view(request, form_url, extra_context)
 
+    def change_view(self, request, object_id, form_url='', extra_context=None):
+        if request.POST and request.POST.get('config-0-device') and \
+                request.POST.get('organization'):
+            device = Device.objects.get(pk=request.POST.get('config-0-device'))
+            org = Organization.objects.get(pk=request.POST.get('organization'))
+            device.organization = org
+            device.save()
+        return super(DeviceAdmin, self).change_view(request, object_id, form_url, extra_context)
```

Comment:
why is this necessary? I don't understand. This shouldn't be necessary. Try to remove it and if it fails for some reason, let the build fail so I can understand what is wrong and help you out.

Reply:
For some reason which i could not put my finger on, the `instance.device.organization` in
```
invalids = templates.exclude(organization=instance.device.organization) \
                                .exclude(organization=None) \ 
                                .values('name')
```
is the organization of the old device and not the new organization that is being set by the user. this is even the mean reason why an inconsistent state is achieve. for example if the old organization of device was `org1`, when you change it to `org2` this instance still holds `org1` instead of `org2` thus, i used this `change_view` to make sure the organization of instance will always be the present organization in this case `org2` instead of `org1